### PR TITLE
fix: removed redundant model properties from the conversation

### LINKED ISSUE #114: Model type changed for the conversation after reload
1. Create new conversation. Check in local storage - coversation.model.type - **'appication'**
2. Reload the page. Check in local storage - coversation.model.type - **'model'** should be **'appication'** -\> **bug**

image

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
Issue #114 Model type changed for the conversation after reload

Removed all model properties except **id** in the **conversation**.

Refs: #114

### CODE REVIEW

**Comment by denys-kolomiitsev** [comment]
/deploy-review

>Deployment status: success

File: src/components/Chat/AssistantSubModelSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  assistantModelId: string;
+  onSelectAssistantSubModel: (modelId: string) => void;
+}
+
+export const AssistantSubModelSelector = ({
+  assistantModelId,
+  onSelectAssistantSubModel,
+}: Props) => {
+  const { t } = useTranslation('chat');
+  const modelsModels = useAppSelector(ModelsSelectors.selectModelsOnly);
```
Comment (mikitabut) [inline src/components/Chat/AssistantSubModelSelector.tsx]: modelsModels? i think onlyModels will be better :)

File: src/components/Chat/AssistantSubModelSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const modelsMap = useAppSelector(ModelsSelectors.selectModelsMap);
+  const assistantSubModel = useMemo(
+    () => modelsMap[assistantModelId],
+    [assistantModelId, modelsMap],
+  );
+  return (
```
Comment (mikitabut) [inline src/components/Chat/AssistantSubModelSelector.tsx:30]: nit: add empty row before

File: src/store/conversations/conversations.epics.ts [hunk 3]
```diff
@@ -187,6 +187,10 @@ const exportConversationsEpic: AppEpic = (action$, state$) =>
 const importConversationsEpic: AppEpic = (action$, state$) =>
   action$.pipe(
     filter(ConversationsActions.importConversations.match),
+    map(({ payload }) => ({
+      payload,
+      modelsMap: ModelsSelectors.selectModelsMap(state$.value),
```
Comment (mikitabut) [inline src/store/conversations/conversations.epics.ts]: You don't need to have it here

File: src/components/Chat/AssistantSubModelSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <SettingContainer>
```
Comment (mikitabut) [inline src/components/Chat/AssistantSubModelSelector.tsx]: Let's move out this setting container as you did for another settings and use empty fragment here instead <></>

File: src/components/Chat/Chat.tsx [hunk 5]
```diff
@@ -285,9 +285,10 @@ export const Chat = memo(() => {
 
   const handleSelectModel = useCallback(
     (conversation: Conversation, modelId: string) => {
-      const newAiEntity = models.find(
-        ({ id }) => id === modelId,
-      ) as OpenAIEntityModel;
+      const newAiEntity = modelsMap[modelId];
+      if (!newAiEntity) {
```
Comment (mikitabut) [inline src/components/Chat/Chat.tsx:289]: is it real case?

**Review by mikitabut — CHANGES_REQUESTED** [review]

**Review by irinakartun — DISMISSED** [review]

File: src/components/Chat/AssistantSubModelSelector.tsx (hunk 1 above)
Comment (denys-kolomiitsev) [reply src/components/Chat/AssistantSubModelSelector.tsx]: Thank you for suggestion.

File: src/store/conversations/conversations.epics.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply src/store/conversations/conversations.epics.ts]: I are right )
Removed.

File: src/components/Chat/AssistantSubModelSelector.tsx (hunk 2 above)
Comment (denys-kolomiitsev) [reply src/components/Chat/AssistantSubModelSelector.tsx:30]: row added

File: src/components/Chat/AssistantSubModelSelector.tsx (hunk 4 above)
Comment (denys-kolomiitsev) [reply src/components/Chat/AssistantSubModelSelector.tsx]: Changes added as suggested.

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: success

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Deployment status: success

**Review by mikitabut — APPROVED** [review]

**Pull request merged**